spec/src/main/resources/META-INF/smithy/bsp/bsp.smithy

```diff
@@ -0,0 +1,1380 @@
+    compileProvider: CompileProvider
+    /// The languages the server supports test execution via method buildTarget/test.
+    testProvider: TestProvider
+    /// The languages the server supports run via method buildTarget/run.
+    runProvider: RunProvider
+    /// The languages the server supports debugging via method debugSession/start.
+    debugProvider: DebugProvider
+    /// The server can provide a list of targets that contain a
+    /// * single text document via the method buildTarget/inverseSources
+    inverseSourcesProvider: Boolean = false
+    /// The server provides sources for library dependencies
+    /// via method buildTarget/dependencySources
+    dependencySourcesProvider: Boolean = false
+    /// The server can provide a list of dependency modules (libraries with meta information)
+    /// via method buildTarget/dependencyModules
+    dependencyModulesProvider: Boolean = false
+    /// The server provides all the resource dependencies
+    /// via method buildTarget/resources
+    resourcesProvider: Boolean = false
+    /// The server provides all output paths
+    /// via method buildTarget/outputPaths
+    outputPathsProvider: Boolean = false
+    /// The server sends notifications to the client on build
+    /// target change events via buildTarget/didChange
+    buildTargetChangedProvider: Boolean = false
+    /// The server can respond to `buildTarget/jvmRunEnvironment` requests with the
+    /// necessary information required to launch a Java process to run a main class.
+    jvmRunEnvironmentProvider: Boolean = false
+    /// The server can respond to `buildTarget/jvmTestEnvironment` requests with the
+    /// necessary information required to launch a Java process for testing or
+    // debugging.
+    jvmTestEnvironmentProvider: Boolean = false
+    /// Reloading the build state through workspace/reload is supported
+    canReload: Boolean = false
+}
+
+@mixin
+structure LanguageProvider {
+    @required
+    languageIds: LanguageIds
+}
+
+structure CompileProvider with [LanguageProvider] {
+
+}
+
+structure RunProvider with [LanguageProvider] {
+
+}
+
+structure DebugProvider with [LanguageProvider] {
+
+}
+
+structure TestProvider with [LanguageProvider] {
+
+}
+
+structure InitializedBuildParams {
+
+}
+
+@mixin
+structure MessageParams {
+    /// the message type.
+    @required
+    type: MessageType
+    /// The task id if any.
+    task: TaskId
+    /// The request id that originated this notification.
+    /// The originId field helps clients know which request originated a notification in case several requests are handled by the
+    /// client at the same time. It will only be populated if the client defined it in the request that triggered this notification.
+    originId: RequestId
+    /// The actual message.
+    @required
+    message: String
+}
+
+structure OnBuildShowMessageParams with [MessageParams] {
+
+}
+
+structure LogMessageParams with [MessageParams] {
+
+}
+
+@enumKind("open")
```

Comment:
Should this be open?

Reply:
Your prerogative, but this is faithful to the current spec. If you wanted a closed enumeration, the typescript should have looked like : 

``` 
type MessageType = 1 | 2 | 3 | 4
```

Reply:
Yea, go with what we currently have then. To be honest I'm not 100% sure what is there now was actually mean to be that way, open, or if it was just an oversight. Either way, go with what exists now.